src/main/java/com/iwaconsolti/school/demo/service/CourseService.java

```diff
@@ -0,0 +1,57 @@
+package com.iwaconsolti.school.demo.service;
+
+import com.iwaconsolti.school.demo.model.Course;
+import com.iwaconsolti.school.demo.model.Student;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+@Service
+@Slf4j
+public class CourseService {
+
+    private final List<Course> courses = new ArrayList<>();
+
+    public CourseService() {
+        Course mathematics = Course.builder().id(1).name("Matematicas").professorName("Sergio").creationDate(new Date()).build();
+        Course Spanish = Course.builder().id(2).name("Español").professorName("Guadalupe").creationDate(new Date()).build();
+
+        this.courses.add(mathematics);
+        this.courses.add(Spanish);
+    }
+
+    public List<Course> getCourses() {
+        return this.courses;
+    }
+
+    public Course getById(int id) {
+        log.info("Id: {}", id);
+        List<Course> courses = this.courses.stream().filter(course -> course.getId() == id).toList();
+        log.info("Students: {}", courses);
+        if(!courses.isEmpty())
+            return courses.getFirst();
+        return null;
+    }
+
+    public Course createCourse(Course course){
+        if(course != null){
+            course.setCreationDate(new Date());
+            this.courses.add(course);
+        }
+        return course;
+    }
+
+    public Course update(Course course){
+        this.courses.stream().forEach(c -> {
```

Comment:
You could first find the course and if that is found then update the values and return that object. That way you dont have to iterate the list twice